### PR TITLE
Fwmt 430

### PR DESCRIPTION
**Pull request by spdiaz1000**
more Tests

### CODE REVIEW

File: build.gradle [hunk 1]
```diff
@@ -135,7 +136,17 @@ jacocoTestReport {
         xml.enabled true
         html.enabled false
     }
-}
+    afterEvaluate {
```
Comment (soundargit) [inline build.gradle]: Do we need this if we can exclude codecov.yml?

File: src/test/java/uk/gov/ons/fwmt/job_service/data/csv_parser/CSVIteratorTests.java [hunk 2]
```diff
@@ -0,0 +1,44 @@
...
+import org.junit.runner.RunWith;
+import org.mockito.Mock;
+import org.mockito.junit.MockitoJUnitRunner;
+
+import uk.gov.ons.fwmt.job_service.exceptions.types.FWMTCommonException;
+
+import static org.mockito.Mockito.when;
+
+@RunWith(MockitoJUnitRunner.class)
+public class CSVIteratorTests {
```
[the remark this answers is not in the document]
Comment (spdiaz1000) [reply src/test/java/uk/gov/ons/fwmt/job_service/data/csv_parser/CSVIteratorTests.java]: deleted

File: src/main/java/uk/gov/ons/fwmt/job_service/service/impl/FileIngestServiceImpl.java [hunk 3]
```diff
@@ -31,18 +32,19 @@
...
   public SampleSummaryDTO validateSampleFile(File file) throws IOException{
     SampleFilenameComponents filename = SampleFileUtils.buildSampleFilenameComponents(file);
     final Reader reader = new InputStreamReader(new FileInputStream(file), StandardCharsets.UTF_8);
     
-    Iterator<CSVParseResult<LegacySampleIngest>> csvRowIterator = csvParsingService.parseLegacySample(reader, filename.getTla());
+    Iterator<CSVParseResult<LegacySampleIngest>> csvRowIterator = CSVParserBuilder.buildLegacySampleParserIterator(reader, filename.getTla(), fieldPeriodResourceServiceClient);
+    return valitadateCSVRows(file.getName(), csvRowIterator);
+  }
+
+  protected SampleSummaryDTO valitadateCSVRows(String filename, Iterator<CSVParseResult<LegacySampleIngest>> csvRowIterator) {
```
[the remark this answers is not in the document]
Comment (spdiaz1000) [reply src/main/java/uk/gov/ons/fwmt/job_service/service/impl/FileIngestServiceImpl.java]: done

File: src/main/java/uk/gov/ons/fwmt/job_service/data/csv_parser/CSVParserBuilder.java [hunk 4]
```diff
@@ -0,0 +1,30 @@
...
+import uk.gov.ons.fwmt.job_service.data.legacy_ingest.LegacySampleSurveyType;
+import uk.gov.ons.fwmt.job_service.rest.client.FieldPeriodResourceServiceClient;
+
+@Slf4j
+public class CSVParserBuilder {
+  public static Iterator<CSVParseResult<LegacySampleIngest>> buildLegacySampleParserIterator(Reader reader,
+      LegacySampleSurveyType legacySampleSurveyType, FieldPeriodResourceServiceClient fieldPeriodResourceServiceClient) throws IOException {
+    log.debug("Began parsing file: surveyType={}", legacySampleSurveyType);
+    CSVParser parser = getCSVFormat().parse(reader);
+    Iterator<CSVParseResult<LegacySampleIngest>> i = new LegacySampleIterator(parser, legacySampleSurveyType, fieldPeriodResourceServiceClient);
```
[the remark this answers is not in the document]
Comment (spdiaz1000) [reply src/main/java/uk/gov/ons/fwmt/job_service/data/csv_parser/CSVParserBuilder.java]: I find them useful when I debug code

File: src/main/java/uk/gov/ons/fwmt/job_service/data/dto/SampleSummaryDTO.java [hunk 5]
```diff
@@ -1,13 +1,15 @@
...
 import lombok.AllArgsConstructor;
+import lombok.Builder;
 import lombok.Data;
 import uk.gov.ons.fwmt.job_service.data.csv_parser.UnprocessedCSVRow;
 
 import java.util.List;
 
 @Data
 @AllArgsConstructor
+@Builder
```
[the remark this answers is not in the document]
Comment (spdiaz1000) [reply src/main/java/uk/gov/ons/fwmt/job_service/data/dto/SampleSummaryDTO.java]: used in Tests to create objects quickly

File: src/main/java/uk/gov/ons/fwmt/job_service/data/file_ingest/SampleFilenameComponents.java [hunk 6]
```diff
@@ -1,18 +1,23 @@
...
-import java.time.LocalDateTime;
-
 /**
  * This class describes a filename as seen by the FileIngesterService, split into constituent parts
  */
+
 @Data
 @AllArgsConstructor
+@NoArgsConstructor
+@Builder
```
[the remark this answers is not in the document]
Comment (spdiaz1000) [reply src/main/java/uk/gov/ons/fwmt/job_service/data/file_ingest/SampleFilenameComponents.java]: used in Tests to create objects quickly

**Pull request merged**